lib/query-interface.d.ts

```diff
@@ -55,46 +56,45 @@ export interface QueryOptions {
    */
   useMaster?: boolean;
 
-  /**
-   * A function that gets executed while running the query to log the sql.
-   */
-  logging?: Function
-
   /**
    * A sequelize instance used to build the return instance
    */
   instance?: Model;
+}
+
 
+export interface QueryOptionsWithModel {
   /**
    * A sequelize model used to build the returned model instances (used to be called callee)
    */
-  model?: typeof Model;
-
-  // TODO: force, cascade
+  model: typeof Model;
+}
 
+export interface QueryOptionsWithType {
+  /**
+   * The type of query you are executing. The query type affects how results are formatted before they are
+   * passed back. The type is a string, but `Sequelize.QueryTypes` is provided as convenience shortcuts.
+   */
+  type: 'SELECT' | 'INSERT' | 'UPDATE' | 'DELETE' | string;
```

Comment:
If it allows `string`, this type is not any more typesafe than `string`. Same as anything `| any`...

Reply:
True, not sure why this is in there, I'll make that use the new enum type instead.